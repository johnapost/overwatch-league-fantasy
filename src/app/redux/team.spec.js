import reducer, {
  defaultState,
  teamDraftSelect,
  teamDraftPlace,
  teamMoveSelect,
  teamMovePlace,
  teamSetDrafter
} from "./team";

describe("team", () => {
  const player = {
    givenName: "Nina",
    familyName: "Williams",
    headshot: "https://url.url/nina-williams.jpg",
    name: "OWLName",
    nationality: "FI",
    id: "1234",
    teamId: 4523
  };

  describe("action creators", () => {
    it("teamDraftSelect should return action", () => {
      expect(teamDraftSelect(player)).toMatchSnapshot();
    });

    it("teamDraftPlace should return action", () => {
      expect(teamDraftPlace(0)).toMatchSnapshot();
    });

    it("teamMoveSelect should return action", () => {
      expect(teamMoveSelect(0)).toMatchSnapshot();
    });

    it("teamMovePlace should return action", () => {
      expect(teamMovePlace(1)).toMatchSnapshot();
    });

    it("teamSetDrafter should return action", () => {
      expect(teamSetDrafter(12345)).toMatchSnapshot();
    });
  });

  describe("reducer", () => {
    it("should return default state", () => {
      expect(reducer(undefined, { type: "" })).toMatchSnapshot();
    });

    it("should update state when drafting players", () => {
      const draftSelect = reducer(
        {
          ...defaultState,
          roster: {
            0: null,
            1: null
          }
        },
        teamDraftSelect(player)
      );
      expect(draftSelect).toMatchSnapshot();

      const draftPlace = reducer(draftSelect, teamDraftPlace(1));
      expect(draftPlace).toMatchSnapshot();
    });

    it("should update state when assigning drafters", () => {
      const firstDrafter = reducer(defaultState, teamSetDrafter(12345));
      expect(firstDrafter).toMatchSnapshot();

      const secondDrafter = reducer(firstDrafter, teamSetDrafter(54321));
      expect(secondDrafter).toMatchSnapshot();
    });

    it("should update state when moving players", () => {
      const moveSelect = reducer(
        {
          ...defaultState,
          roster: {
            0: player,
            1: null
          }
        },
        teamMoveSelect(0)
      );
      expect(moveSelect).toMatchSnapshot();

      const movePlace = reducer(moveSelect, teamMovePlace(1));
      expect(movePlace).toMatchSnapshot();
    });
  });
});
