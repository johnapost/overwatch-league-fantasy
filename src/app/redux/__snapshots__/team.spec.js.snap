// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`team action creators teamDraftPlace should return action 1`] = `
Object {
  "toIndex": 0,
  "type": "DRAFT_PLACE",
}
`;

exports[`team action creators teamDraftSelect should return action 1`] = `
Object {
  "player": Object {
    "familyName": "Williams",
    "givenName": "Nina",
    "headshot": "https://url.url/nina-williams.jpg",
    "id": "1234",
    "name": "OWLName",
    "nationality": "FI",
    "teamId": 4523,
  },
  "type": "DRAFT_SELECT",
}
`;

exports[`team action creators teamMovePlace should return action 1`] = `
Object {
  "toIndex": 1,
  "type": "MOVE_PLACE",
}
`;

exports[`team action creators teamMoveSelect should return action 1`] = `
Object {
  "fromIndex": 0,
  "type": "MOVE_SELECT",
}
`;

exports[`team action creators teamSetDrafter should return action 1`] = `
Object {
  "type": "SET_DRAFTER",
  "uid": 12345,
}
`;

exports[`team reducer should return default state 1`] = `
Object {
  "drafter": null,
  "roster": Object {},
  "selectedPlayer": null,
}
`;

exports[`team reducer should update state when assigning drafters 1`] = `
Object {
  "drafter": 12345,
  "roster": Object {},
  "selectedPlayer": null,
}
`;

exports[`team reducer should update state when assigning drafters 2`] = `
Object {
  "drafter": 54321,
  "roster": Object {},
  "selectedPlayer": null,
}
`;

exports[`team reducer should update state when drafting players 1`] = `
Object {
  "drafter": null,
  "roster": Object {
    "0": null,
    "1": null,
  },
  "selectedPlayer": Object {
    "familyName": "Williams",
    "givenName": "Nina",
    "headshot": "https://url.url/nina-williams.jpg",
    "id": "1234",
    "name": "OWLName",
    "nationality": "FI",
    "teamId": 4523,
  },
}
`;

exports[`team reducer should update state when drafting players 2`] = `
Object {
  "drafter": null,
  "roster": Object {
    "0": null,
    "1": Object {
      "familyName": "Williams",
      "givenName": "Nina",
      "headshot": "https://url.url/nina-williams.jpg",
      "id": "1234",
      "name": "OWLName",
      "nationality": "FI",
      "teamId": 4523,
    },
  },
  "selectedPlayer": null,
}
`;

exports[`team reducer should update state when moving players 1`] = `
Object {
  "drafter": null,
  "roster": Object {
    "0": Object {
      "familyName": "Williams",
      "givenName": "Nina",
      "headshot": "https://url.url/nina-williams.jpg",
      "id": "1234",
      "name": "OWLName",
      "nationality": "FI",
      "teamId": 4523,
    },
    "1": null,
  },
  "selectedPlayer": Object {
    "familyName": "Williams",
    "givenName": "Nina",
    "headshot": "https://url.url/nina-williams.jpg",
    "id": "1234",
    "name": "OWLName",
    "nationality": "FI",
    "teamId": 4523,
  },
}
`;

exports[`team reducer should update state when moving players 2`] = `
Object {
  "drafter": null,
  "roster": Object {
    "0": null,
    "1": Object {
      "familyName": "Williams",
      "givenName": "Nina",
      "headshot": "https://url.url/nina-williams.jpg",
      "id": "1234",
      "name": "OWLName",
      "nationality": "FI",
      "teamId": 4523,
    },
  },
  "selectedPlayer": null,
}
`;
